### PR TITLE
[oneDNN] Shape inference fix for int8 convolutions.

### PR DESCRIPTION
**Pull request by Srini511**
In current implementation of shape inference for convolution(fused with padding), padding dimensions are calculated only if explicit padding list is provided. For int8 convolutions (with fused paddings) we provide paddings through a padding list which is the same as explicit paddings but a different name. This PR includes padding list also into padding calculation if present.

### CODE REVIEW

File: tensorflow/core/framework/common_shape_fns.cc [hunk 1]
```diff
@@ -707,7 +706,18 @@ Status Conv2DShapeImpl(shape_inference::InferenceContext* c,
...
   } else {
+    std::vector<int64_t> p_list;
+    Status s_p_list = c->GetAttr("padding_list", &p_list);
     CHECK(padding != Padding::EXPLICIT);  // Crash ok.
+    if (!s_p_list.ok() && !errors::IsNotFound(s_p_list)) {
+      return s_p_list;
+    }
+    if (s_p_list.ok() && !p_list.empty()) {
+      padding = Padding::EXPLICIT;
+      explicit_paddings = p_list;
```
Comment (penpornk) [inline tensorflow/core/framework/common_shape_fns.cc:717]: Please add comments explaining that fused int8 conv ops use this field to support explicit padding.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

Is `padding_list` only used in oneDNN ops (and vanilla TF ops that are placeholders for oneDNN ops)? Would it be possible to just change its name to `explicit_padding` to be consistent with other standard TF ops?

**Comment by Srini511** [comment]
> Thank you for the PR!
> 
> Is `padding_list` only used in oneDNN ops (and vanilla TF ops that are placeholders for oneDNN ops)? Would it be possible to just change its name to `explicit_padding` to be consistent with other standard TF ops?

Yes AFAIK it is used only in oneDNN int8 ops and the TF placeholders. Regarding naming, it would need changes to op signature, quantization tools like INC and possibly other places. I can discuss internally and let you know of the scope for that.
I have made the changes requested. Please take a look.

**Review by penpornk — APPROVED** [review]
Thank you very much for the changes and I'm so sorry for the delay!

**Pull request merged**